src/formatDate.js

```diff
@@ -8,7 +8,37 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const [yearKey, monthKey, dayKey, separator] = fromFormat;
+  const splittedDate = date.split(separator);
+
+  const oldDateObj = [yearKey, monthKey, dayKey].reduce((acc, key, index) => {
+    acc[key] = splittedDate[index];
+
+    return acc;
+  }, {});
+
+  const updatedDateObj = changeYearsFormat(oldDateObj, toFormat);
```

Comment:
The function `changeYearsFormat` is well-named, indicating its purpose, but it's mutating the input object. It would be better to return a new object instead of modifying the existing one to avoid side effects.